jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketChannelState.java

```diff
@@ -19,140 +19,197 @@
 package org.eclipse.jetty.websocket.core.internal;
 
 import org.eclipse.jetty.websocket.core.CloseStatus;
-
-import java.util.concurrent.atomic.AtomicReference;
+import org.eclipse.jetty.websocket.core.Frame;
+import org.eclipse.jetty.websocket.core.OpCode;
+import org.eclipse.jetty.websocket.core.ProtocolException;
 
 /**
  * Atomic Connection State
  */
 public class WebSocketChannelState
 {
-    private static class State
+    enum State
     {
-        final String name;
-        final boolean inOpen;
-        final boolean outOpen;
-        final CloseStatus closeStatus;
-
-        State(String name, boolean inOpen, boolean outOpen, CloseStatus closeStatus)
-        {
-            this.name = name;
-            this.inOpen = inOpen;
-            this.outOpen = outOpen;
-            this.closeStatus = closeStatus;
-        }
-
-        @Override
-        public String toString()
-        {
-            return String.format("%s{i=%b o=%b c=%d}", name, inOpen, outOpen, closeStatus == null?-1:closeStatus.getCode());
-        }
+        CONNECTING,
+        CONNECTED,
+        OPEN,
+        ICLOSED,
+        OCLOSED,
```

Comment:
ISHUT and OSHUT are better - they match with any other socket API / state that we have in Jetty.